### PR TITLE
fill_mask helper

### PR DESCRIPTION
**Pull request by julien-c**

### CODE REVIEW

**Comment by LysandreJik** [comment]
This feels like a very similar method to [generate](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_utils.py#L567). If we want to add it to the library's internals, don't you think it would make sense to add it directly to the models' internals like `generate`? I feel it would be consistent to mirror it but for mask filling.

This way it could be called like this:

```py
model.fill_mask(x)
```

Furthermore, I think that handling strings is nice, but handling lists of tokens would be better, like `generate` does -> more coherent with `generate`, only handle model data (tokens) and no need to pass a tokenizer to a `modeling_utils` internal method.

**Comment by julien-c** [comment]
Ok I'll take a deeper look next week @LysandreJik

**Comment by julien-c** [comment]
Ok, I see what you mean. 

Our desired use case is that we would be able to do this in one line, on a string. It's a usage sample we've been seeing more and more (initially from RoBERTa), as it lets one check that the LM works well, in one line.

Examples:
- https://github.com/musixmatchresearch/umberto#fairseq-1
- https://camembert-model.fr/

I see two options:
- define it in its own utils-like file, e.g. `fill_mask.py` or `hub_utils.py` or whatever
- create a `FillMaskPipeline`, as the pipeline already has a tokenizer and a model

Wdyt?

**Comment by LysandreJik** [comment]
I'm hyped by the pipeline option. I believe filling a mask, being the main use-case of an MLM trained model would be a very nice pipeline to have, alongside sequence classification, question answering and named entity recognition.

**Comment by julien-c** [comment]
I think you are very right and I wholeheartedly agree with you. <3 <3

**Review by mfuntowicz — APPROVED** [review]
It looks very cool 👍 ! 

As a future note, if we include more models (using other tokenizers) we may want to expose the mask_token property so that the user doesn't have to know about the model internals to feed the right mask token.

File: tests/test_pipelines.py [hunk 1]
```diff
@@ -62,9 +63,25 @@
     )
 }
 
+FILL_MASK_FINETUNED_MODELS = {
+    ("distilroberta-base", "distilroberta-base", None),
```
Comment (LysandreJik) [inline tests/test_pipelines.py]: Is there a reason there's only distilroberta here?

File: src/transformers/pipelines.py [hunk 2]
```diff
@@ -491,6 +506,71 @@ def __call__(self, *args, **kwargs):
...
+                logits = outputs[i, masked_index, :]
+                probs = logits.softmax(dim=0)
+                values, predictions = probs.topk(self.topk)
+
+            for v, p in zip(values.tolist(), predictions.tolist()):
+                tokens = input_ids.numpy()
+                tokens[masked_index] = p
+                # Filter padding out:
+                tokens = tokens[np.where(tokens != self.tokenizer.pad_token_id)]
+                result.append({"sequence": self.tokenizer.decode(tokens), "score": v, "token": p})
```
Comment (LysandreJik) [inline src/transformers/pipelines.py]: clean

**Review by LysandreJik — APPROVED** [review]

**Comment by LysandreJik** [comment]
Looks good to me, you can merge

**Comment by julien-c** [comment]
@mfuntowicz As the pipeline exposes its tokenizer, I'm guessing you can already do something like (untested):

```python
fill_mask = pipeline("fill-mask")
fill_mask(f"My name is {fill_mask.tokenizer.mask_token}.")
```

**Comment by thomwolf** [comment]
Nice one @julien-c!
